framework/workspace/internal/workspacemanager.cpp

```diff
@@ -53,78 +166,103 @@ void WorkspaceManager::init()
         });
     }
 
+    configuration()->currentWorkspaceName().ch.onReceive(this, [this](const std::string&) {
+        setupCurrentWorkspace();
+    });
+
     load();
 }
 
 void WorkspaceManager::load()
 {
     m_workspaces.clear();
 
-    std::vector<io::path> files = findWorkspaceFiles();
-    for (const io::path& f : files) {
-        auto w = std::make_shared<Workspace>(f);
-        m_workspaces.push_back(w);
+    io::paths files = findWorkspaceFiles();
+
+    for (const io::path& file : files) {
+        auto workspace = std::make_shared<Workspace>(file);
+        m_workspaces.push_back(workspace);
     }
 
     setupCurrentWorkspace();
 }
 
-std::vector<io::path> WorkspaceManager::findWorkspaceFiles() const
+io::paths WorkspaceManager::findWorkspaceFiles() const
 {
-    std::vector<io::path> files;
-    std::vector<io::path> paths = configuration()->workspacePaths();
-    for (const io::path& path : paths) {
-        //! TODO Change on use IFileSystem
-        QStringList flist = QDir(path.toQString()).entryList({ "*.workspace" }, QDir::Files);
-        for (const QString& f : flist) {
-            files.push_back(path + "/" + f);
+    io::paths result;
+    io::paths dirPaths = configuration()->workspacePaths();
+
+    for (const io::path& dirPath : dirPaths) {
+        RetVal<io::paths> files = fileSystem()->scanFiles(dirPath, { "*.workspace" });
+        if (!files.ret) {
+            LOGE() << files.ret.toString();
+            continue;
         }
+
+        result.insert(result.end(), files.val.begin(), files.val.end());
     }
-    return files;
+
+    return result;
 }
 
 void WorkspaceManager::setupCurrentWorkspace()
 {
-    std::string wsname = configuration()->currentWorkspaceName();
+    saveCurrentWorkspace();
 
-    std::shared_ptr<Workspace> w = findAndInit(wsname);
-    if (!w) {
-        LOGW() << "filed get workspace: " << wsname << ", will use Default";
-        w = findAndInit(std::string(DEFAULT_WORKSPACE_NAME));
-    }
+    std::string workspaceName = configuration()->currentWorkspaceName().val;
 
-//    IF_ASSERT_FAILED(w) {
-//        return;
-//    }
+    WorkspacePtr workspace = findAndInit(workspaceName);
+    if (!workspace) {
+        LOGW() << "failed get workspace: " << workspaceName << ", will use Default";
```

Comment:
better use DEFAULT_WORKSPACE_NAME instead of Default

Reply:
fixed